src/Microsoft.ML.Data/Transforms/FeatureContributionCalculationTransform.cs

```diff
@@ -0,0 +1,367 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using Microsoft.ML.Core.Data;
+using Microsoft.ML.Data;
+using Microsoft.ML.Runtime;
+using Microsoft.ML.Runtime.CommandLine;
+using Microsoft.ML.Runtime.Data;
+using Microsoft.ML.Runtime.EntryPoints;
+using Microsoft.ML.Runtime.Internal.Internallearn;
+using Microsoft.ML.Runtime.Internal.Utilities;
+using Microsoft.ML.Runtime.Model;
+
+[assembly: LoadableClass(FeatureContributionCalculatingTransformer.Summary, typeof(FeatureContributionCalculatingTransformer), null, typeof(SignatureLoadModel),
+    FeatureContributionCalculatingTransformer.FriendlyName, FeatureContributionCalculatingTransformer.LoaderSignature)]
+
+[assembly: LoadableClass(typeof(IRowMapper), typeof(FeatureContributionCalculatingTransformer), null, typeof(SignatureLoadRowMapper),
+   FeatureContributionCalculatingTransformer.FriendlyName, FeatureContributionCalculatingTransformer.LoaderSignature)]
+
+[assembly: LoadableClass(typeof(void), typeof(FeatureContributionEntryPoint), null, typeof(SignatureEntryPointModule), FeatureContributionCalculatingTransformer.LoaderSignature)]
+
+namespace Microsoft.ML.Runtime.Data
+{
+    /// <summary>
+    /// The FeatureContributionCalculationTransformer computes model-specific contribution scores for each feature.
+    /// See the list of currently supported predictors below.
+    /// </summary>
+    /// <remarks>
+    /// Feature Contribution Calculation is currently supported for the following Predictors:
+    ///     Regression:
+    ///         OrdinaryLeastSquares, StochasticDualCoordinateAscent (SDCA), OnlineGradientDescent, PoissonRegression,
+    ///         GeneralizedAdditiveModels (GAM), LightGbm, FastTree, FastForest, FastTreeTweedie
+    ///     Binary Classification:
+    ///         AveragedPerceptron, LinearSupportVectorMachines, LogisticRegression, StochasticDualCoordinateAscent (SDCA),
+    ///         StochasticGradientDescent (SGD), SymbolicStochasticGradientDescent, GeneralizedAdditiveModels (GAM),
+    ///         FastForest, FastTree, LightGbm
+    ///     Ranking:
+    ///         FastTree, LightGbm
+    ///
+    /// See the sample below for an example of how to compute feature importance using the FeatureContributionCalculatingTransformer.
+    /// </remarks>
+    /// <example>
+    /// <format type="text/markdown">
+    /// <![CDATA[
+    /// [!code-csharp[FCT](~/../docs/samples/docs/samples/Microsoft.ML.Samples/Dynamic/FeatureContributionCalculationTransform.cs)]
+    /// ]]>
+    /// </format>
+    /// </example>
+    public sealed class FeatureContributionCalculatingTransformer : RowToRowTransformerBase
```

Comment:
>RowToRowTransformerBase [](start = 68, length = 23)

why not one to one?

Reply:
Yes now that we removed the other code I can make it one to one! good point

---
In reply to: [243094570](https://github.com/dotnet/machinelearning/pull/1847#discussion_r243094570) [](ancestors = 243094570)